### PR TITLE
Custom storage can be used in Karl.

### PR DESCRIPTION
**Pull request by mzhurovich**
@dkorolev Moar metaprogramming!

### CODE REVIEW

File: Karl/test.cc [hunk 1]
```diff
@@ -905,6 +906,82 @@ TEST(Karl, EndToEndTest) {
...
+    const custom_karl_t karl(storage, UnittestKarlParameters());
+    const current::karl::Locator karl_locator(Printf("http://localhost:%d/", FLAGS_karl_test_keepalives_port));
+    const karl_unittest::ServiceGenerator generator(
+        FLAGS_karl_generator_test_port, std::chrono::microseconds(1000), karl_locator);
+    generator_codename = generator.ClaireCodename();
+
+    const auto result =
+        karl.InternalExposeStorage().ReadOnlyTransaction([&](ImmutableFields<custom_storage_t> fields) {
+          ASSERT_TRUE(Exists(fields.claires[generator_codename]));
+          EXPECT_EQ(current::karl::ClaireRegisteredState::Active,
```
Comment (dkorolev) [inline Karl/test.cc]: These would be unreadable in the console output if the test fails, right?

**Comment by dkorolev** [comment]
LGTM++!

File: Karl/test.cc (hunk 1 above)
Comment (mzhurovich) [inline Karl/test.cc]: Readable enough IMHO:
```
Value of: Value(fields.claires[generator_codename]).registered_state
  Actual: 1-byte object <01>
Expected: current::karl::ClaireRegisteredState::Active
Which is: 1-byte object <00>
```

Comment (dkorolev) [inline Karl/test.cc]: Merging!

**Pull request merged**